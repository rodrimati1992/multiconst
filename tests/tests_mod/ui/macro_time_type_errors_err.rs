// ensure that mismatched syntactic tuple pattern are type error at macro expansion time
multiconst::multiconst!{
    const (A200, A201): () = ();
}

// ensure that mismatched syntactic tuple pattern are type errors
multiconst::multiconst!{
    const (): (u32, u32) = ();
}

// ensure that array patterns don't work with tuple types
multiconst::multiconst!{
    const [A]: (u32,) = [10];
}

// ensure that tuple patterns don't work with array types
multiconst::multiconst!{
    const (A,): [u32; 1] = [10];
}







fn main(){}