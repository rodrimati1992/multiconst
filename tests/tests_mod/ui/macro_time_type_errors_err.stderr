error: tuple pattern has 2 elements, but type has 0
 --> tests/tests_mod/ui/macro_time_type_errors_err.rs:3:11
  |
3 |     const (A200, A201): () = ();
  |           ^^^^^^^^^^^^
  |
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: tuple pattern has 0 elements, but type has 2
 --> tests/tests_mod/ui/macro_time_type_errors_err.rs:8:11
  |
8 |     const (): (u32, u32) = ();
  |           ^^
  |
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: mismatched pattern and type
  --> tests/tests_mod/ui/macro_time_type_errors_err.rs:13:11
   |
13 |     const [A]: (u32,) = [10];
   |           ^^^
   |
   = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error: mismatched pattern and type
  --> tests/tests_mod/ui/macro_time_type_errors_err.rs:18:11
   |
18 |     const (A,): [u32; 1] = [10];
   |           ^^^^
   |
   = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)
